packages/mediapipe-core/lib/src/io/native_memory_manager.dart

```diff
@@ -0,0 +1,112 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ffi';
+import 'package:ffi/ffi.dart';
+import 'package:mediapipe_core/io.dart';
+
+/// {@template NativeMemoryManager}
+/// Container for native memory fully managed by Dart code, but hydrated by
+/// native code. Exists to pair the correct allocation and deallocation
+/// functionality for a given data type.
+///
+/// See also:
+///  * [NativeStringManager] for typical usage.
+/// {@endtemplate}
+abstract class NativeMemoryManager<T extends NativeType> {
```

Comment:
You're right, I'm aiming for a mixture of `1` and `2` / `2.1`. (Where I have the `2.2` scenario, I'm already going down the direction you indicated.)

I'm guessing that `2.1` is preferable to `1`, since it's lighter weight and, as you said, composable with `Allocator`s, so I'll move forward converting to extension types which free the memory.